roles/network_plugin/calico/templates/calico-node.yml.j2

```diff
@@ -223,6 +226,8 @@ spec:
 #            # Configure the IP Pool from which Pod IPs will be chosen.
 #            - name: CALICO_IPV4POOL_CIDR
 #              value: "{{ calico_pool_cidr | default(kube_pods_subnet) }}"
+            - name: NO_DEFAULT_POOLS
```

Comment:
Is this one a side change or normal ?

Reply:
Calico extracts the pod_subnet from the configmap: `kubeadm-config` and creates a default IP pool (provided the variable CALICO_IPV4POOL_CIDR is not empty, it is empty in kubespray). see https://github.com/projectcalico/calico/blob/aff70be8ba03e1d8b2ea3b35372afa3910717ea4/node/pkg/lifecycle/startup/startup.go#L745

But kubespray does not create a default IP pool in this way, by creating it manually. see https://github.com/kubernetes-sigs/kubespray/blob/a1521dc16e268f116cf428cf497cb0433c2fa66e/roles/network_plugin/calico/tasks/install.yml#L244. So if so, should we make sure that calico doesn't create a default IP pool? 

I'm not sure, unless I'm missing something.

Reply:
In addition, the names of the default calico's IPPools created by kubespray are `default-pool` and `default-pool-v6`. It would be nice if we are aligned with the calico community.

https://github.com/projectcalico/calico/blob/aff70be8ba03e1d8b2ea3b35372afa3910717ea4/node/pkg/lifecycle/startup/startup.go#L58)